File: Javascript/src/DSA/stack.js
"use strict";
import { LinkedList } from "./linked-list";

export class Stack {
    constructor() {
        this.count = 0;
        this.list = new LinkedList();
    }
    push(value) {
        this.list.insert(value, this.count);
        this.count++;
    }
    pop() {
        this.list.removeAt(this.count - 1);
        this.count--;
    }
    printStack() {
        return this.list.printList();
    }
}
let stack = new Stack();
console.log("inserting values");
stack.push(1);
console.log(stack.printStack());
stack.push(2);
console.log(stack.printStack());
stack.push(3);
console.log(stack.printStack());
stack.push(4);
console.log(stack.printStack());
console.log("removing values");
stack.pop();
console.log(stack.printStack());
stack.pop();
console.log(stack.printStack());
stack.pop();
console.log(stack.printStack());
stack.pop();
console.log(stack.printStack());
stack.pop();
console.log(stack.printStack());
stack.pop();
console.log(stack.printStack());

File: C#/DSA/Stack.cs
using DSA.util;
using System.Collections;

namespace DSA
{
    public class Stack<T>
    {
        private readonly ArrayList array;
        private int count;

        public Stack()
        {
            array = new ArrayList();
            count = 0;
        }

        public void Push(T element)
        {
            array.Add(element);
            count++;
        }

        public void Pop()
        {
            array.RemoveAt(count - 1);
            count--;
        }

        public int Size()
        {
            return count;
        }

        public bool IsEmpty()
        {
            return count == 0;
        }

        public T Peek()
        {
            return (T)array[count - 1];
        }

        public override string ToString()
        {
            return JSon.ConvertToJson(array);
        }
    }
}
